### PR TITLE
go-modules: fix build, module name and local imports

### PR DESCRIPTION
**Pull request by neolit123**
fix the build in current master:

- update module to github.com/kubernetes-sigs/cluster-api-provider-docker
- modify github.com/chuckha/cluster-api-provider-docker imports to
github.com/kubernetes-sigs/cluster-api-provider-docker/kind/actions
run 'go mod tidy'
- rename DeleteClusterNode to DeleteNode
the existing functions don't have Cluster in the name:
DeleteControlPlane, DeleteWorker

/kind bug
/priority backlog
/assign @chuckha

### CODE REVIEW

File: kind/actions/kind.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func DeleteControlPlane(clusterName, nodeName string) error {
 
 	// pick the first one, but there should never be multiples since docker requires name to be unique.
 	node := nodeList[0]
-	if err := DeleteClusterNode(clusterName, nodeName); err != nil {
+	if err := DeleteNode(clusterName, nodeName); err != nil {
```
Comment (pablochacin) [inline kind/actions/kind.go]: This change is unrelated to the purpose of the PR. Is intentional to add it here?

**Comment by neolit123** [comment]
updated OP to include the reason for the rename.
without it the build is broken for kind-test.

File: kind/actions/cluster_actions.go [hunk 2]
```diff
@@ -306,8 +306,8 @@ func GetNodeRefUID(clusterName, nodeName string) (string, error) {
 	return strings.TrimSpace(lines[0]), nil
 }
 
-// DeleteClusterNode will remove the kubernetes node from the list of nodes (during a kubectl get nodes).
-func DeleteClusterNode(clusterName, nodeName string) error {
+// DeleteNode will remove the kubernetes node from the list of nodes (during a kubectl get nodes).
```
Comment (chuckha) [inline kind/actions/cluster_actions.go]: I have a PR out that fixes this issue, this isn't quite right since the delete action became two different things to work properly. DeleteClusterNode is the thing that both DeleteWorker and DeleteControlPlane call after they do the steps necessary for deleting a worker and deleting a control plane

Comment (neolit123) [reply kind/actions/cluster_actions.go]: without the delete rename, the build fails for me.

Comment (chuckha) [reply kind/actions/cluster_actions.go]: I have a PR out that uses the correct methods in kind-test.

Comment (chuckha) [reply kind/actions/cluster_actions.go]: https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/14

Comment (neolit123) [reply kind/actions/cluster_actions.go]: ok, found https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/14

**Comment by neolit123** [comment]
updated to exclude the Delete* rename.
